security-analysis/security-analysis-api/src/main/java/com/powsybl/security/detectors/AbstractLimitViolationDetector.java

```diff
@@ -75,9 +80,17 @@ public void checkCurrent(Contingency contingency, Branch branch, Consumer<LimitV
         checkCurrent(contingency, branch, Branch.Side.TWO, consumer);
     }
 
+    @Override
+    public void checkCurrent(Contingency contingency, ThreeWindingsTransformer transformer, Consumer<LimitViolation> consumer) {
+        checkCurrent(contingency, transformer, ThreeWindingsTransformer.Side.ONE, consumer);
+        checkCurrent(contingency, transformer, ThreeWindingsTransformer.Side.TWO, consumer);
+        checkCurrent(contingency, transformer, ThreeWindingsTransformer.Side.THREE, consumer);
+    }
+
     @Override
     public void checkAll(Contingency contingency, Network network, Consumer<LimitViolation> consumer) {
         network.getBranchStream().forEach(b -> checkCurrent(contingency, b, consumer));
+        network.getThreeWindingsTransformerStream().forEach(t -> checkCurrent(contingency, t, consumer));
```

Comment:
It seems this is not tested for three windings transformer, I think there is very big test on this on SecurityAnalysisTest using EurostagTutorialExample1Factory, and also smaller test on LimitViolationDetectorTest using EurostagTutorialExample1Factory with contingencies, do we need this "contingency feature" for three windings transformer?

Reply:
Yes, we want to offer the same possibilities for three windings transformers as for branches.

The default limit detector doesn't consider the contingencies, but some custom implementations can take them into account for branches and it would be a problem if this could not be also done for three windings transformers.

Reply:
done